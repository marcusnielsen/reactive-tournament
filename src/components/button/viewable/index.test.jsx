import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Provider from '../../provider'
import Model, {variants} from '../../button'
import Button from './index'

test('create viewable', () => {
  const provider = Provider()
  const model = Model({
    provider,
    name: 'a',
    variant: variants.default
  })
  const viewable = Button({model})
  expect(Object.keys(viewable))
  .toEqual([
    'colorTheme',
    'labels',
    'actions',
    'state_',
    'viewState_',
    'pureView',
    'view'
  ])
})

test('render view', () => {
  const provider = Provider()
  const model = Model({
    provider,
    name: 'a',
    variant: variants.default
  })
  const viewable = Button({model})
  const View = viewable.pureView

  // @TODO: Get hold of the viewState_ here or we need to recombine.
  return viewable.viewState_.take(1).forEach((state) => {
    const mountWrapper = shallow(<View state={state} />)
    const component = mountWrapper.first().shallow()
    expect(toJson(component)).toMatchSnapshot()
  })
})
