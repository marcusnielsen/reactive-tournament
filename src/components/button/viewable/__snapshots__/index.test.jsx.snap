exports[`test render view 1`] = `
<button
  className="fWekty"
  onClick={[Function]}
  tabIndex="-1">
  a
</button>
`;
